### PR TITLE
 feat: revamp code snippets tutorial

### PR DESCRIPTION
**Pull request by p7nov**
Rewrite the tutorial [Run code snippets](https://kotlinlang.org/docs/run-code-snippets.html) to show actual options: scratches and worksheets in the IDE, online playground in browser, ki shell in the command line.

### CODE REVIEW

**Comment by p7nov** [comment]
[WIP]: figuring out a better place and name for this page. Content is ready for review.

**Comment by p7nov** [comment]
@dromanov-jb, @ligee could you check the parts about the playground and ki-shell correspondingly?

File: docs/topics/run-code-snippets.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 
-* [Scratches](#scratches-and-worksheets) let us write and run code in a temporary file outside of our project in the IDE.
-* [Worksheets](#scratches-and-worksheets) are like scratches, but they reside within projects.
-* [REPL](#repl) (_Read-Eval-Print-Loop_) runs code in an interactive console.     
+* [Scratches and worksheets](#intellij-idea-scratches-and-worksheets) in IntelliJ IDEA
+* [Kotlin Playground](#browser-kotlin-playground) in the browser
+* [ki shell](#command-line-ki-shell) in the command line
 
-## Scratches and worksheets
+## IntelliJ IDEA: Scratches and worksheets
```
Comment (etolstoy) [inline docs/topics/run-code-snippets.md:11]: AFAIK AndroidStudio also supports scratches. Shouldn’t we mention it?

File: docs/topics/run-code-snippets.md [hunk 2]
```diff
@@ -55,30 +55,135 @@ To rebuild the module automatically before each run of a scratch or a worksheet,
 
 ![Scratch select module](scratch-select-module.png){width=700}
 
-### Running as REPL 
+### Run as REPL 
 
-To evaluate each particular expression in a scratch or a worksheet, run it with **Use REPL** selected. The code will be 
-executed the same way as in [REPL](#repl): the code lines will run sequentially, providing results of each call. 
+To evaluate each particular expression in a scratch or a worksheet, run it with **Use REPL** selected. The code lines
```
Comment (etolstoy) [inline docs/topics/run-code-snippets.md:60]: @yalishevant Do you remember the results of our discussions about «Run as REPL» feature? Do we plan to deprecate it?

**Review by etolstoy — APPROVED** [review]

File: docs/topics/run-code-snippets.md (hunk 1 above)
Comment (p7nov) [reply docs/topics/run-code-snippets.md:11]: Yeah, it does. Let's change this to `IDE` and mention Android Studio in this section.

**Comment by ligee** [comment]
About the `ki-shell` example session snapshots:
first, the latest version (0.4.5) fixed Slf4j warnings, so the output is a bit cleaner.
and second - the version of ki contains `SNAPSHOT` substring, that doesn't seem like a good signal to me. I asked maintainers to remove it in the next release.
So, it makes sense to wait until the next release and remake snapshots.

File: docs/topics/run-code-snippets.md [hunk 3]
```diff
@@ -55,30 +55,135 @@ To rebuild the module automatically before each run of a scratch or a worksheet,
...
+
+Along with the standard library, ki shell supports external dependencies.
+This lets you use it for trying third-party libraries without creating a whole project. 
+
+To add a third-party library in ki shell, use the `:dependsOn` command. By default, ki shell works with Maven Central,
+but you can use other repositories if you connect them with the `:repository` command:
+
+![ki shell external dependency](ki-shell-dependency.png){width=700}
+
+### Extensibility
```
Comment (ligee) [inline docs/topics/run-code-snippets.md:181]: I'm not sure if we want to advertise the plugin authoring yet. I would skip this section and refer to the ki repository for further details (without mentioning extensibility here at all).

Comment (p7nov) [reply docs/topics/run-code-snippets.md]: Thanks. Removed the part about extensibility and added the GitHub link explicitly in the intro.

**Comment by p7nov** [comment]
[WIP] again: waiting for the new release of ki shell with clearer output to make screenshots.

**Comment by p7nov** [comment]
> About the `ki-shell` example session snapshots: first, the latest version (0.4.5) fixed Slf4j warnings, so the output is a bit cleaner. and second - the version of ki contains `SNAPSHOT` substring, that doesn't seem like a good signal to me. I asked maintainers to remove it in the next release. So, it makes sense to wait until the next release and remake snapshots.

@ligee I've made new screenshots on ki shell 0.4.5 ( without `-SNAPSHOT` and slf4j warnings)

File: docs/topics/run-code-snippets.md (hunk 2 above)
Comment (yalishevant) [reply docs/topics/run-code-snippets.md:60]: Yes, we _decided_ to remove this check-box. Run as REPL will be the only way of running.
We heaven't planned it for a particular time and person who will do this. Roma Golyshev is now estimating the scope of work.

Comment (p7nov) [reply docs/topics/run-code-snippets.md:60]: So we're leaving it as is, I guess? I'll keep in mind the future update. Could you share the corresponding YT issue or add me to watchers?

**Review by ligee — APPROVED** [review]
Looks great to me!

File: docs/topics/run-code-snippets.md [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
 
-* [Scratches](#scratches-and-worksheets) let us write and run code in a temporary file outside of our project in the IDE.
-* [Worksheets](#scratches-and-worksheets) are like scratches, but they reside within projects.
-* [REPL](#repl) (_Read-Eval-Print-Loop_) runs code in an interactive console.     
+* [Scratches and worksheets](#ide-scratches-and-worksheets) in the IDE
+* [Kotlin Playground](#browser-kotlin-playground) in the browser
+* [ki shell](#command-line-ki-shell) in the command line
 
-## Scratches and worksheets
+## IDE: scratches and worksheets
```
[the remark this answers is not in the document]
Comment (koshachy) [reply docs/topics/run-code-snippets.md:11]: Not a mistake, just trying to find a better visual representation for a text.

File: docs/topics/run-code-snippets.md [hunk 5]
```diff
@@ -55,30 +55,125 @@ To rebuild the module automatically before each run of a scratch or a worksheet,
...
+[Kotlin Playground](https://play.kotlinlang.org/) is an online application that allows writing, running, and sharing
+Kotlin code in the browser.
 
-To run REPL in IntelliJ IDEA, open **Tools** | **Kotlin** | **Kotlin REPL**.
+### Write and edit code
 
-To run REPL in the OS command line, open `/bin/kotlinc-jvm` from the directory of standalone Kotlin compiler.
+In the Playground editor area, you can write code as you would write in a source file:
+* place your own classes, functions, and top-level declarations in an arbitrary order
+* write the executable part in the body of the `main` function.
```
Comment (koshachy) [inline docs/topics/run-code-snippets.md:75]:
```suggestion
* write the executable part in the body of the `main()` function.
```

File: docs/topics/run-code-snippets.md [hunk 6]
```diff
@@ -55,30 +55,125 @@ To rebuild the module automatically before each run of a scratch or a worksheet,
...
 
-REPL supports multiline input as well. The result of the multiline input is the value of its last expression. 
+The Playground highlights the code and shows code completion options as you type. It automatically imports declarations
+from the standard library and [`kotlinx.coroutines`](coroutines-overview.md).
 
-![REPL multi-line run](repl-multi-line.png){width=700}
+### Choose execution environment
 
+The Playground provides means for customizing the execution environment:
+* multiple Kotlin versions, including available previews of a future version
```
Comment (koshachy) [inline docs/topics/run-code-snippets.md:88]: Link to the EAP? 
Or it could just distract user? 🤔

**Review by koshachy — APPROVED** [review]
Nice job.
Left some comments.

File: docs/topics/run-code-snippets.md (hunk 6 above)
Comment (p7nov) [reply docs/topics/run-code-snippets.md:88]: Yeah, why not. Thanks for the idea

File: docs/topics/run-code-snippets.md [hunk 7]
```diff
@@ -55,30 +55,125 @@ To rebuild the module automatically before each run of a scratch or a worksheet,
...
+
+When you press **Enter**, ki shell evaluates the entered line and prints the result. Expression values are
+printed as variables with auto-generated names like `res*`. You can later use such variables in the code you run. 
+If the entered construct is incomplete (for example, an `if` with a condition but without the body), the shell prints 
+three dots and expects the remaining part.
+
+
+![ki shell results](ki-shell-results.png){width=700}
+
+Type in `:help` (or `:h`) to see commands available in ki shell. Below is a quick overview of some helpful ones.
```
[the remark this answers is not in the document]
Comment (p7nov) [reply docs/topics/run-code-snippets.md:151]: Yes, let's keep it shorter

File: docs/topics/run-code-snippets.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (p7nov) [reply docs/topics/run-code-snippets.md:11]: Arranged the definitions in a list

**Pull request merged**